### PR TITLE
Implement Salesforce 'Account' and "Opportunity' actions

### PR DESCRIPTION
**Pull request by LLuque-twilio**
_A summary of your pull request, including the what change you're making and why._

This PR addresses [STRATCONN 1033](https://segment.atlassian.net/browse/STRATCONN-1033) and [STRATCONN 1038](https://segment.atlassian.net/browse/STRATCONN-1038) JIRA tickets.

It introduces the 'Account' and 'Opportunity' actions to the Salesforce Destination. It includes the three functionalities create, update, and upsert for both actions.

## Testing

_Include any additional information about the testing you have completed to
ensure your changes behave as expected. For a speedy review, please check
any of the tasks you completed below during your testing._

- [x] Added [unit tests](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing) for new functionality
Screen Shot 2022-02-23 at 4 46 43 PM
Screen Shot 2022-02-23 at 4 47 07 PM
- [x] Tested end-to-end using the [local server](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing)
- [x] [Segmenters] Tested in the staging environment
Screen Shot 2022-02-23 at 4 37 31 PM
Screen Shot 2022-02-23 at 4 41 09 PM

### CODE REVIEW

File: packages/destination-actions/src/destinations/salesforce/sf-object-to-shape.ts [hunk 1]
```diff
@@ -20,9 +27,42 @@ const CaseShape = (payload: GenericPayload): CaseBaseShapeType => {
...
+    Phone: payload.phone,
+    Description: payload.description,
+    Website: payload.website
+  }
+}
+
 const objectToShape = new Map<string, (payload: GenericPayload) => LeadBaseShapeType | CaseBaseShapeType>([
   ['lead', LeadShape],
-  ['case', CaseShape]
+  ['case', CaseShape],
```
Comment (nick-Ag) [inline packages/destination-actions/src/destinations/salesforce/sf-object-to-shape.ts:63]: Did line 61 give you type errors without including `OpprotunityBaseShapeType | AccountBaseShapeType` ?

Comment (LLuque-twilio) [reply packages/destination-actions/src/destinations/salesforce/sf-object-to-shape.ts:63]: Great point completely overlooked this and no it didn't... super interesting. I will add them in anyways just for alignment.

Comment (LLuque-twilio) [reply packages/destination-actions/src/destinations/salesforce/sf-object-to-shape.ts:66]: Just tested it, and removing LeadBaseShapeType and replacing with Account/Opportunity baseShapeTypes did throw an error... Not sure why.

Comment (nick-Ag) [reply packages/destination-actions/src/destinations/salesforce/sf-object-to-shape.ts:66]: Hmm weird, thanks for adding them in

**Review by nick-Ag — APPROVED** [review]
Looks good :+1: 🔥

**Pull request merged**